### PR TITLE
Staked Well

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

 This PR adds a Staked Well integration.

### CODE REVIEW

File: contracts/src/instances/stk-well/StkWellTarget4.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+pragma solidity 0.8.22;
+
+import { HyperdriveTarget4 } from "../../external/HyperdriveTarget4.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IHyperdriveAdminController } from "../../interfaces/IHyperdriveAdminController.sol";
+import { StkWellBase } from "./StkWellBase.sol";
+
+/// @author DELV
+/// @title StkWellTarget4
+/// @notice StkWellHyperdrive's target3 logic contract. This contract contains
```
Comment (jrhea) [inline contracts/src/instances/stk-well/StkWellTarget4.sol:11]:
```suggestion
/// @notice StkWellHyperdrive's target4 logic contract. This contract contains
```

File: contracts/src/interfaces/IStkWell.sol [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { IERC4626 } from "./IERC4626.sol";
+
+// ****************************************************************************
+// FIXME: Fill out the interface as needed.  ERC4626 extended as an example.
```
Comment (jrhea) [inline contracts/src/interfaces/IStkWell.sol:7]: do we need this FIXME?

File: test/utils/InstanceTest.sol [hunk 3]
```diff
@@ -1,6 +1,9 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.20;
 
+// FIXME
+import { console2 as console } from "forge-std/console2.sol";
```
Comment (jrhea) [inline test/utils/InstanceTest.sol:5]: remove console.log

**Review by jrhea — APPROVED** [review]
lgtm just a couple of nits

File: contracts/src/instances/stk-well/StkWellHyperdrive.sol [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+/// SSSSSSSS|_______________|______________||_______________|______________|SSSSSSSS
+/// SSSSSSSS                                                                SSSSSSSS
+/// SSSSSSSS                                                                SSSSSSSS
+///
+/// @author DELV
+/// @title StkWellHyperdrive
+/// @notice A Hyperdrive instance that uses a StkWell vault as the yield source.
+/// @dev This instance supports the Well staking vault on Base. The vault shares
+///      token is staked well. Some high-level features of this yield source:
+///
```
Comment (mcclurejt) [inline contracts/src/instances/stk-well/StkWellHyperdrive.sol:57]:
```suggestion
```

**Review by mcclurejt — APPROVED** [review]
LGTM

File: contracts/src/instances/stk-well/StkWellHyperdrive.sol (hunk 4 above)
Comment (jalextowle) [reply contracts/src/instances/stk-well/StkWellHyperdrive.sol:57]: This spacing was intentional. We use this spacing in other implementations FWIW.

**Pull request merged**